front_end/content.py

```diff
@@ -1407,18 +1417,24 @@ def save_course(self, course_basics, course_details):
         return course_basics["id"]
 
     def save_assignment(self, assignment_basics, assignment_details):
+        # clean and join allowed_ip_addresses
+        assignment_details["allowed_ip_addresses"][:] = [x for x in assignment_details["allowed_ip_addresses"] if x != "" and x != ","]
```

Comment:
I think this is where you'd want to add the `strip()` call.

I'm not sure if you need the `[:]` on the left side of the equals operator.